src/components/SearchBar.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+    <div class="search-bar">
+        <v-text-field
+            label="Search for dishes here..."
+            v-model="searchtext"
+            :loading="isLoading"
+            @keyup="filterSearch()"
+        >
+        </v-text-field>
+        <button v-if="searchtext != ''" @click="clearFilter">
+            <figure>
+                <v-icon size="24px" >mdi-close</v-icon>
+            </figure>
+        </button>
+    </div>
+</template>
+<script>
+export default {
+    name: 'SearchBar',
+    props: ['hasFiltedItems', 'dishes'],
+    data () {
+        return {
+            searchtext: '',
+            isLoading: false,
```

Comment:
This should come from the store. There is already a isLoadingDishes, isLoadingCourses and a method that returns true if either is still loading. You could use that instead. Create  computed property that returns that from the store